vault/quotas/quotas.go

```diff
@@ -543,6 +569,20 @@ func (m *Manager) SetEnableRateLimitResponseHeaders(val bool) {
 	m.config.EnableRateLimitResponseHeaders = val
 }
 
+// SetRateLimitExemptPaths updates the rate limit exempt paths in the Manager's
+// configuration in addition to updating the path manager. Every call to
+// SetRateLimitExemptPaths will wipe out the existing path manager and set the
+// paths based on the provided argument.
+func (m *Manager) SetRateLimitExemptPaths(vals []string) {
+	if vals == nil {
+		vals = []string{}
+	}
+
+	m.config.RateLimitExemptPaths = vals
+	m.rateLimitPathManager = pathmanager.New()
+	m.rateLimitPathManager.AddPaths(vals)
```

Comment:
I think that you should AddPaths to the new manager before moving it "into production". Even if the write to rateLimitPathManager itself is not a race condition (or is resolved by making it atomic), the following interleaved sequence is possible:

goroutine A: m.rateLimitPathManager = pathManager.New()
goroutine B: ApplyRateLimitQuota to an exempt path, have it rejected because pathManager is empty
goroutine A: AddPaths()

Reply:
I don't think this is necessary as I've added a write-lock to `SetRateLimitExemptPaths`.